assets/app/scripts/extensions/javaLink.js

```diff
@@ -43,7 +43,8 @@ angular.module('openshiftConsoleExtensions', ['openshiftConsole'])
       if (!pod || !pod.status || pod.status.phase !== 'Running') {
         return;
       }
-      var podName = pod.metadata.name;
+      // always use https when connecting to jolokia in a pod
+      var podName = 'https:' + pod.metadata.name;
```

Comment:
this should move into `ProxyPod`. When specifying the scheme, the port always has to be provided, even if empty, so you'd need to do `https:name:` for an empty port